common/src/gas_provider/internal.rs

```diff
@@ -768,6 +775,44 @@ where
         node.lock().ok_or_else(|| InternalError::forbidden().into())
     }
 
+    fn reserve(
```

Comment:
the internal logic of the function looks much like in `create_from_with_value`. Better adjust the latter one and call it in the body of the `reserve`.